trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/TestProxiedAssumedRoleRequests.java

```diff
@@ -46,14 +47,15 @@ public class TestProxiedAssumedRoleRequests
     private final TestingCredentialsController credentialsController;
 
     @Inject
-    public TestProxiedAssumedRoleRequests(TestingHttpServer httpServer, @ForTesting Credentials testingCredentials, TestingCredentialsController credentialsController)
+    public TestProxiedAssumedRoleRequests(TestingHttpServer httpServer, @ForTesting Credentials testingCredentials, TestingCredentialsController credentialsController, @ForS3MockContainer S3Client storageClient)
     {
+        super(storageClient);
         this.httpServer = requireNonNull(httpServer, "httpServer is null");
         this.testingCredentials = requireNonNull(testingCredentials, "testingCredentials is null");
         this.credentialsController = credentialsController;
     }
 
-    @AfterEach
+    @AfterAll
     public void validateCount()
     {
         assertThat(credentialsController.assumedRoleCount()).isGreaterThan(0);
```

Comment:
I'm really not happy with this assumedRole count :(

Reply:
I'm happy to change it - but we'd need to have some sort of instrumented S3 Client. The number of times it'll assume roles depends on the number of operations we run against the proxy - which is test dependent